coalib/coala_main.py

```diff
@@ -71,31 +74,43 @@ def run_coala(log_printer=None,
         dtag = str(sections['default'].get('dtag', None))
         config_file = os.path.abspath(str(sections["default"].get("config")))
 
-        # Deleting all .orig files, so the latest files are up to date!
-        coala_delete_orig.main(log_printer, sections["default"])
-
-        delete_tagged_results(dtag, config_file, log_printer)
-
-        for section_name, section in sections.items():
-            if not section.is_enabled(targets):
-                continue
-
-            print_section_beginning(section)
-            section_result = execute_section(
-                section=section,
-                global_bear_list=global_bears[section_name],
-                local_bear_list=local_bears[section_name],
-                print_results=print_results,
-                log_printer=log_printer)
-            yielded, yielded_unfixed, results[section_name] = (
-                simplify_section_result(section_result))
-
-            yielded_results = yielded_results or yielded
-            yielded_unfixed_results = (
-                yielded_unfixed_results or yielded_unfixed)
-            did_nothing = False
-
-            file_dicts[section_name] = section_result[3]
+        settings_hash = get_settings_hash(sections)
+        flush_cache = sections['default'].get('flush_cache', False)
+        if not flush_cache and settings_changed(log_printer, settings_hash):
+            log_printer.warn("Since the configuration settings have been "
+                             "changed since the last run, the "
+                             "cache will be flushed and rebuilt.")
```

Comment:
I think the FileCache or settings_changed should do a log message and here you can just do `flush_cache = sections['default'].get('flush_cache', False) or settings_changed(log_printer, settings_hash)`